ingest/src/main/java/com/microsoft/azure/kusto/ingest/IngestionProperties.java

```diff
@@ -33,7 +42,54 @@ public IngestionReportMethod getReportMethod() {
         return reportMethod;
     }
 
-    public Map<String, String> getAdditionalProperties() {
+    public ArrayList<String> getDropByTags() {
+        return dropByTags;
+    }
+
+    public ArrayList<String> getIngestByTags() {
+        return ingestByTags;
+    }
+
+    public ArrayList<String> getAdditionalTags() {
+        return additionalTags;
+    }
+
+    public ArrayList<String> getIngestIfNotExists() {
+        return ingestIfNotExists;
+    }
+
+    public Map<String, String> getAdditionalProperties() throws IOException {
+        if(!dropByTags.isEmpty() || !ingestByTags.isEmpty() || !additionalTags.isEmpty()){
+            ArrayList<String> tags = new ArrayList<>();
+            if (!additionalTags.isEmpty())
+            {
+                tags.addAll(additionalTags);
+            }
+            if (!ingestByTags.isEmpty())
+            {
+                for(String t : ingestByTags){
+                    tags.add(String.format("%s%s", "drop-by:", t));
+                }
+            }
+            if (!dropByTags.isEmpty())
+            {
+                for(String t : ingestByTags){
+                    tags.add(String.format("%s%s", "ingest-by:", t));
+                }
+            }
+
+            ObjectMapper objectMapper = new ObjectMapper();
+            String tagsAsJson = objectMapper.writeValueAsString(tags);
+            additionalProperties.put("tags", tagsAsJson);
+        }
+
+        if (!ingestIfNotExists.isEmpty())
+        {
+            ObjectMapper objectMapper = new ObjectMapper();
+            String ingestIfNotExistsJson = objectMapper.writeValueAsString(ingestIfNotExists);
+            additionalProperties.put("ingestIfNotExists", ingestIfNotExistsJson );
+        }
+
         return additionalProperties;
```

Comment:
I wonder - if the get method calculates the value every time from scratch and there's no direct set method, is there're a point in saving it?